atomkraft/test/test.py

```diff
@@ -40,9 +41,22 @@ def __init__(
         Initialize a test for a given trace, which can be a directory or a file.
         """
         self.root = root
-        self.name = Test.make_name(trace)
-        self.dir = Test.make_dir(root, trace.is_dir(), self.name)
-        self.file_path = self.dir / f"test_{self.name}.py"
+        self.trace = trace
+        self.timestamp = snakecase(datetime.now().isoformat(timespec="milliseconds"))
+
+        if trace.is_dir():
+            self.name = Test._path_to_id(trace)
+            self.tests_dir = root / "tests" / f"{self.name}_{self.timestamp}"
+        else:
+            self.name = Test._path_to_id(trace) + "_" + self.timestamp
+            self.tests_dir = root / "tests"
+        self.tests_dir.mkdir(parents=True, exist_ok=True)
+        self.file_path = self.tests_dir / f"test_{self.name}.py"
+
+        self.report_dir = (
+            root / "reports" / f"{Test._path_to_id(trace)}_{self.timestamp}"
+        )
```

Comment:
It makes sense to organize the directory here. how about `root / "reports" / f"{Test._path_to_id(trace)}" / self.timestamp`?

Reply:
Yes, it's better. Done.